rir/src/compiler/translations/pir_2_rir/pir_2_rir.cpp

```diff
@@ -70,9 +70,8 @@ class SSAAllocator {
 
     void computeStackAllocation() {
 
-        // For now, just flip a coin on where to put it
         static auto toStack = [](Instruction* i) -> bool {
-            return Phi::Cast(i) || !MkEnv::Cast(i);
+            return i->numberOfUses() < 2;
```

Comment:
I am a bit worried that you also change the allocation heuristics in the same commit. I would like to see that it works now with the old heuristics, and then change them later.